### DISCUSSION TITLE
handle Decimals with FocalPointPicker ?

### DISCUSSION THREAD

**Discussion by Nonimart** [WordPress/gutenberg#60043] [Q&A]
Hi everyone, 

I'm reposting a question originally submitted [here](https://wordpress.org/support/topic/handle-decimals-with-focalpointpicker/).
(with the idea here might be a better place for that kind of question ? ) .  

I was wondering if there is a propper/native way to handle decimals for the focal point picker component ?

I’am building a simple gutenberg block, requiring precise positionning, with decimals (ex 88.5%) . I could surely handle this with custom fields, but I’m curious to know it if there would be a native way to handle this scenario.

```Javascript 
<FocalPointPicker
   url={coverUrl ?? null}
   value={objectPosition}
   onChange={(focalPoint) => {
   handleChangeobjectPosition(focalPoint);
   }}
   resolvePoint={resolvepoint}
 /> 
```
 
I’ve tried using the resolvePoint method mentionned in the doc, which outputs a precise position, but the number with decimals then get rounded when passed to the handleChange function anyway.

<img src="https://cdn.discordapp.com/attachments/946816770008842240/1219595715261763625/image.png?ex=660be005&is=65f96b05&hm=ac35833f27127a49659a8c2a156f60d725f1c31a7e3b16b6083771e864edb3ab&" />

Many thanks in advance for the Help,

**Comment by stokesman** [comment]
I suspect it may work to side-step the issue using `resolvePoint` to stash the point in a ref at the precision you desire and then reference that from the `onChange` callback instead of the value it receives. 

```javascript
const refPrecisePoint = useRef();

<FocalPointPicker
   url={coverUrl ?? null}
   value={objectPosition}
   onChange={() => {
      handleChangeobjectPosition(refPrecisePoint.current);
   }}
   resolvePoint={(focalPoint) => {
      const [ x, y ] = Object.values( focalPoint ).map(
         // Clamps the value from zero to one and rounds to three decimals.
         v => Math.round( Math.max( 0, Math.min( v, 1 ) * 1e3 ) ) / 1e3
      );
      refPrecisePoint.current = { x, y };
      return focalPoint;
   }}
 /> 
```

If that doesn’t work, it’s going to require opening a feature request that `FocalPointPicker` support configurable precision. Even if this does work it’s a bit ugly/wasteful so the feature request is not frivolous IMO.
